code/datums/diseases/wizarditis.dm

```diff
@@ -9,106 +9,177 @@
 	viable_mobtypes = list(/mob/living/carbon/human)
 	disease_flags = CAN_CARRY|CAN_RESIST|CURABLE
 	spreading_modifier = 0.75
-	desc = "Some speculate that this virus is the cause of the Space Wizard Federation's existence. Subjects affected show the signs of brain damage, yelling obscure sentences or total gibberish. On late stages subjects sometime express the feelings of inner power, and, cite, 'the ability to control the forces of cosmos themselves!' A gulp of strong, manly spirits usually reverts them to normal, humanlike, condition."
+	desc = "Some speculate that this virus is the cause of the Space Wizard Federation's existence. \
+		Subjects affected show the signs of brain damage, yelling obscure sentences or total gibberish. \
+		On late stages subjects sometime express the feelings of inner power, and cite \
+		'the ability to control the forces of cosmos themselves!' \
+		A gulp of strong, manly spirits usually reverts them to normal, humanlike, condition. \
+		A form of magical grounding can help, too, but will not cure it on its own."
 	severity = DISEASE_SEVERITY_HARMFUL
 	required_organs = list(/obj/item/bodypart/head)
 
-/*
-BIRUZ BENNAR
-SCYAR NILA - teleport
-NEC CANTIO - dis techno
-EI NATH - shocking grasp
-AULIE OXIN FIERA - knock
-TARCOL MINTI ZHERI - forcewall
-STI KALY - blind
-*/
+	/// List of random non-targeted spells to pick from to cast
+	var/list/datum/action/cooldown/spell/random_spells = list()
+	/// List of random targeted spells to pick from to cast
+	var/list/datum/action/cooldown/spell/random_targeted_spells = list()
+
+	/// The hat type to give the infected
+	var/hat_type
+	/// The robe type to give the infected
+	var/robe_type
+
+/datum/disease/wizarditis/after_add()
+	switch(pick("blue", "red", "yellow"))
+		if("blue")
+			hat_type = /obj/item/clothing/head/wizard
+			robe_type = /obj/item/clothing/suit/wizrobe
+		if("red")
+			hat_type = /obj/item/clothing/head/wizard/red
+			robe_type = /obj/item/clothing/suit/wizrobe/red
+		if("yellow")
+			hat_type = /obj/item/clothing/head/wizard/yellow
+			robe_type = /obj/item/clothing/suit/wizrobe/yellow
+
+	init_spells()
+
+/datum/disease/wizarditis/Destroy()
+	QDEL_LIST(random_spells)
+	QDEL_LIST(random_targeted_spells)
+	return ..()
 
 /datum/disease/wizarditis/stage_act(seconds_per_tick, times_fired)
 	. = ..()
 	if(!.)
 		return
 
+	if(affected_mob.can_block_magic(charge_cost = 0))
+		update_stage(1)
+		return
+
+	if(stage >= 3 && SPT_PROB(0.15 * stage, seconds_per_tick))
+		var/datum/action/cooldown/spell/picked = pick(random_spells)
+		if(!picked.try_invoke(affected_mob, feedback = FALSE))
+			to_chat(affected_mob, span_danger("You feel something building up inside... but the feeling passes."))
+			return
+
+		picked.spell_feedback(affected_mob)
+		return
+
+	if(stage <= 3 && SPT_PROB(0.33 * stage, seconds_per_tick))
+		affected_mob.manual_emote("sniffles.")
```

Comment:
Incredibly minor, but can you put the '<= 3' code before the '>= 3' code? That way they're still ordered by stage

Reply:
No, the earlier proc early returns if they do a fake invocations, which means they can't sniffle.
They shouldn't sniffle at the same time as they're doing an invocation (which can happen exactly at stage 3).

Reply:
Sad, fair enough